src/java/org/apache/cassandra/index/sai/disk/v1/PartitionAwarePrimaryKeyFactory.java

```diff
@@ -100,19 +100,39 @@ public Clustering clustering()
 
         @Override
         public ByteSource asComparableBytes(ByteComparable.Version version)
+        {
+            return asComparableBytes(version == ByteComparable.Version.LEGACY ? ByteSource.END_OF_STREAM : ByteSource.TERMINATOR, version, true);
+        }
+
+        @Override
+        public ByteSource asComparableBytesMinPrefix(ByteComparable.Version version)
+        {
+            return asComparableBytes(ByteSource.LT_NEXT_COMPONENT, version, false);
+        }
+
+        @Override
+        public ByteSource asComparableBytesMaxPrefix(ByteComparable.Version version)
+        {
+            return asComparableBytes(ByteSource.GT_NEXT_COMPONENT, version, false);
+        }
+
+        private ByteSource asComparableBytes(int terminator, ByteComparable.Version version, boolean includeNullComponent)
         {
             // Note: Unlike row-aware primary keys the asComparable method in for
             // partition aware keys is only used on the write side so we do not need
             // to enforce deferred loading here.
             ByteSource tokenComparable = token.asComparableBytes(version);
             ByteSource keyComparable = partitionKey == null ? null
                                                             :ByteSource.of(partitionKey.getKey(), version);
-            return ByteSource.withTerminator(version == ByteComparable.Version.LEGACY
-                                             ? ByteSource.END_OF_STREAM
-                                             : ByteSource.TERMINATOR,
-                                             tokenComparable,
-                                             keyComparable,
-                                             null);
+
+            if (!includeNullComponent)
```

Comment:
Nit: prefer to handle the true-case in the if when you need both i.e. if (includeNull) { return X } return Y

Reply:
inverted the boolean to `excludeNullComponent`